Update build.zig.zon for zig master/0.14

nihklas opened the pull request:
In a recently landed change (https://github.com/ziglang/zig/pull/22994), the structure of the build.zig.zon file changed. This PR fixes the structure to be compatible with the newest master build and coming 0.14 release.

Currently, the package of upstream tree-sitter points to my fork of it with the same fix. When the upstream package fixed it (https://github.com/tree-sitter/tree-sitter/pull/4258), the package should be updated. For that reason, this PR can only be merged after tree-sitter made it's changes. When that happens I'll update the PR. For now, it can just sit here for other people to see, if they have issues with the newest zig version.

WillLillis commented:
> For that reason, this PR can only be merged after tree-sitter made it's changes. When that happens I'll update the PR.

Happy to merge this in once the updates are made :)

nihklas commented:
I updated the dependency to follow upstream tree-sitter again. The change of the build.zig.zon is not yet in a tagged released, so I updated to url to follow the master of tree-sitter.

WillLillis commented:
Thanks!
